zipkin-receiver-kafka/src/main/scala/com/twitter/zipkin/receiver/kafka/KafkaProcessor.scala

```diff
@@ -1,41 +1,44 @@
 package com.twitter.zipkin.receiver.kafka
 
 import com.twitter.zipkin.thriftscala.{Span => ThriftSpan}
-import kafka.consumer.{Consumer, ConsumerConnector, ConsumerConfig}
-import kafka.serializer.Decoder
 import com.twitter.util.{Closable, CloseAwaitably, FuturePool, Future, Time}
 import java.util.concurrent.{TimeUnit, Executors}
 import java.util.Properties
-import java.net.{SocketAddress, InetSocketAddress}
-
+import kafka.consumer.{Consumer, ConsumerConnector, ConsumerConfig}
+import kafka.serializer.{Decoder, StringDecoder}
 import com.twitter.scrooge.BinaryThriftStructSerializer
 import com.twitter.zipkin.conversions.thrift.{thriftSpanToSpan, spanToThriftSpan}
 import kafka.message.Message
 
+
 object KafkaProcessor {
 
   type KafkaDecoder = Decoder[Option[List[ThriftSpan]]]
 
-  def apply(
+  val defaultKeyDecoder = new StringDecoder()
+
+  def apply[T](
     topics:Map[String, Int],
     config: Properties,
     process: Seq[ThriftSpan] => Future[Unit],
-    decoder: KafkaProcessor.KafkaDecoder
-  ): KafkaProcessor = new KafkaProcessor(topics, config, process, decoder)
+    keyDecoder: Decoder[T],
+    valueDecoder: KafkaDecoder
+  ): KafkaProcessor[T] = new KafkaProcessor(topics, config, process, keyDecoder, valueDecoder)
 }
 
-class KafkaProcessor(
+class KafkaProcessor[T](
```

Comment:
there are many other places where 'T' is used in files below, i'm not going to comment on each of them.